File: app/src/main/java/com/notex/create/ViewActivity.java
package com.notex.create;

import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.view.View;
import android.widget.TextView;
import androidx.appcompat.app.AppCompatActivity;
import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;
import java.lang.reflect.Type;
import java.util.ArrayList;
import java.util.HashMap;

public class ViewActivity extends AppCompatActivity {
	
	private TextView titleText;
	private TextView notebookText;
	private TextView contentText;
	private TextView viewTitleText;
	private SharedPreferences sharedPreferences;
	private Gson gson;
	private ArrayList<HashMap<String, Object>> notes;
	private String viewType;
	private String viewName;
	
	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		setContentView(R.layout.view);
		
		// Initialize views
		titleText = findViewById(R.id.note_title);
		notebookText = findViewById(R.id.notebook_name);
		contentText = findViewById(R.id.note_content);
		viewTitleText = findViewById(R.id.view_title);
		
		// Initialize SharedPreferences and Gson
		sharedPreferences = getSharedPreferences("NoteXData", Context.MODE_PRIVATE);
		gson = new Gson();
		
		// Get view type and name
		viewType = getIntent().getStringExtra("type");
		viewName = getIntent().getStringExtra(viewType.equals("note") ? "title" : "name");
		
		// Load data
		loadData();
		
		// Set up back button
		findViewById(R.id.back_button).setOnClickListener(new View.OnClickListener() {
			@Override
			public void onClick(View v) {
				finish();
			}
		});
		
		// Set up edit button
		findViewById(R.id.edit_button).setOnClickListener(new View.OnClickListener() {
			@Override
			public void onClick(View v) {
				if (viewType.equals("note")) {
					Intent intent = new Intent(ViewActivity.this, EditActivity.class);
					intent.putExtra("title", viewName);
					startActivity(intent);
					finish();
				}
			}
		});
	}
	
	@Override
	protected void onResume() {
		super.onResume();
		loadData(); // Refresh data when returning to this activity
	}
	
	private void loadData() {
		// Load notes
		String notesJson = sharedPreferences.getString("notes", "");
		if (!notesJson.isEmpty()) {
			Type type = new TypeToken<ArrayList<HashMap<String, Object>>>(){}.getType();
			notes = gson.fromJson(notesJson, type);
		} else {
			notes = new ArrayList<>();
		}
		
		if (viewType.equals("note")) {
			// Display single note
			for (HashMap<String, Object> note : notes) {
				if (note.get("title").toString().equals(viewName)) {
					titleText.setText(note.get("title").toString());
					contentText.setText(note.get("content").toString());
					
					String notebook = note.get("notebook").toString();
					if (notebook.isEmpty()) {
						notebookText.setVisibility(View.GONE);
					} else {
						notebookText.setText("Notebook: " + notebook);
						notebookText.setVisibility(View.VISIBLE);
					}
					
					viewTitleText.setText("Note");
					break;
				}
			}
		} else if (viewType.equals("notebook")) {
			// Display notes in notebook
			viewTitleText.setText(viewName);
			titleText.setVisibility(View.GONE);
			notebookText.setVisibility(View.GONE);
			
			StringBuilder content = new StringBuilder();
			for (HashMap<String, Object> note : notes) {
				String notebook = note.get("notebook").toString();
				if (notebook.equals(viewName)) {
					content.append("# ").append(note.get("title").toString()).append("\n\n");
					content.append(note.get("content").toString()).append("\n\n");
					content.append("---\n\n");
				}
			}
			
			contentText.setText(content.toString());
		}
	}
}

File: app/src/main/java/com/notex/create/ViewActivity.kt
package com.notex.create

import android.content.Intent
import android.os.Bundle
import androidx.activity.compose.setContent
import androidx.activity.viewModels
import androidx.appcompat.app.AppCompatActivity
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.Icon
import androidx.compose.material3.IconButton
import androidx.compose.material3.Scaffold
import androidx.compose.material3.Text
import androidx.compose.material3.TopAppBar
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.livedata.observeAsState
import androidx.compose.runtime.remember
import androidx.compose.ui.Modifier
import androidx.compose.ui.res.painterResource
import com.notex.create.viewmodel.NoteViewModel

class ViewActivity : AppCompatActivity() {

    private val noteViewModel: NoteViewModel by viewModels()
    private var viewType: String? = null
    private var viewId: Int = -1

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        viewType = intent.getStringExtra("type")
        viewId = intent.getIntExtra("id", -1)
        setContent {
            ViewScreen(
                noteViewModel = noteViewModel,
                viewType = viewType,
                viewId = viewId,
                onEditClick = {
                    val intent = Intent(this@ViewActivity, EditActivity::class.java)
                    intent.putExtra("id", viewId)
                    startActivity(intent)
                    finish()
                }
            )
        }
    }
}

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun ViewScreen(
    noteViewModel: NoteViewModel,
    viewType: String?,
    viewId: Int,
    onEditClick: () -> Unit
) {
    val notes by noteViewModel.allNotes.observeAsState(initial = emptyList())
    val notebooks by noteViewModel.allNotebooks.observeAsState(initial = emptyList())
    val notesForNotebook by noteViewModel.getNotesByNotebook(viewId).observeAsState(initial = emptyList())

    val note = remember(notes, viewId) { notes.find { it.id == viewId } }
    val notebook = remember(notebooks, viewId) { notebooks.find { it.id == viewId } }

    Scaffold(
        topBar = {
            TopAppBar(
                title = {
                    if (viewType == "note") {
                        Text(note?.title ?: "Note")
                    } else {
                        Text(notebook?.name ?: "Notebook")
                    }
                },
                actions = {
                    if (viewType == "note") {
                        IconButton(onClick = onEditClick) {
                            Icon(painter = painterResource(id = R.drawable.icon_edit_round), contentDescription = "Edit")
                        }
                    }
                }
            )
        }
    ) { paddingValues ->
        Column(
            modifier = Modifier
                .padding(paddingValues)
                .fillMaxSize()
        ) {
            if (viewType == "note") {
                note?.let {
                    Text(text = it.content)
                }
            } else {
                Column {
                    notesForNotebook.forEach { note ->
                        Text(text = "# ${note.title}")
                        Text(text = note.content)
                        Text(text = "---")
                    }
                }
            }
        }
    }
}
